src/Catel.MVVM/Catel.MVVM.Shared/Services/UIVisualizerService.wpf.cs

```diff
@@ -128,40 +128,31 @@ protected virtual void HandleCloseSubscription(object window, object data, Event
         /// <returns><c>true</c> if the window is closed with success; otherwise <c>false</c> or <c>null</c>.</returns>
         protected virtual bool? ShowWindow(FrameworkElement window, bool showModal)
         {
+            Argument.InheritsFrom<System.Windows.Window>("window", window);
+
             if (showModal)
             {
-                var showDialogMethodInfo = window.GetType().GetMethodEx("ShowDialog");
-                if (showDialogMethodInfo != null)
-                {
-                    // Child window does not have a ShowDialog, so not null is allowed
-                    bool? result = null;
+                // Child window does not have a ShowDialog, so not null is allowed
+                bool? result = null;
 
-                    window.Dispatcher.InvokeIfRequired(() =>
+                window.Dispatcher.InvokeIfRequired(() =>
```

Comment:
I believe .Invoke can result in deadlocks (should be BeginInvoke instead). What if we just remove the non-async version?